### ISSUE TITLE
The problem of creating an object with a hierarchy

### ISSUE THREAD

**Issue by q3olegka** [Scirra/Construct-3-bugs#6157]
## Problem description

When creating an instance with a hierarchy, the child is created first, and the parent is show only in the second tick.

In my example, we are creating a 9patch with a text object in the hierarchy. And I can observe on any device that the text is created first and 9patch is already created after it. This happens only after the first download of the game and when the first instance is created

## Attach a .c3p

[hierarchy-issue-r312.zip](https://github.com/Scirra/Construct-3-bugs/files/9690661/hierarchy-issue-r312.zip)

## Steps to reproduce

1. Open the example. (Create a 9patch object, create a text object. Make the text object a child element of the 9patch object)
2. Run the preview. 
3. Press the 1 key. 
4. Keep a close eye on the first created object.

## Observed result

You can notice how after the first launch and when the instance is first created, the text is first created on an empty background, and in the next tick, the 9patch object is already created

## Expected result

At the first launch and the first instance creation, the text should be created in the same tick together with the 9patch object.

## More details

![1](https://user-images.githubusercontent.com/50992459/193417465-37fe8564-6a83-44fa-9246-5dbcc410b3df.jpg)
![2](https://user-images.githubusercontent.com/50992459/193417466-5b8b60bf-ec2a-4302-ab84-5d3d43f33206.jpg)

https://user-images.githubusercontent.com/50992459/193417544-0b1215ad-8894-433c-897f-8de1df7404b6.mp4

In the video, you can see how the problem manifests itself only when I launch the game for the first time.  The next time, even after restarting, the problem does not appear. But, when I close the preview tab and run it again, the problem reappears. This also happens in a particular game - whenever I restart the game on the site, the problem appears

## System details

<details><summary>View details</summary>

PASTE HERE

</details>

**Comment by AshleyScirra** [comment]
Hierarchies are all created at the same time. I believe what you're seeing is the time it takes 9-patch to load its image, because it is not placed in the layout and so Construct will not pre-load it when the layout starts, and creating it will force an on-the-spot dynamic image load that could take a moment to complete.

Place an instance of the 9-patch on the layout and destroy it on startup; then Construct knows that layout will use that object, and preload its image.

**Comment by q3olegka** [comment]
@AshleyScirra 

I still face problems displaying objects after creating them. I store all instances on the same Instance layout, and shouldn't C3 load them in order to create them correctly?

Okay, I put this 9-patch object with its hierarchies on the first layout in the project and it started loading normally. Only one of the sprites in this hierarchy, for some reason, started loading one tick later. 

But that's not all, there are two other sprites that I create in turn, which are also on the Instance layout. And everything worked well until recently. Now the card sprite appears later than the icon sprite. 

The main problem is that the created sprites can be displayed at different times, although they are created in the same event, so it is logical for C3 to display them at the same time and it should take into account objects that are already loaded into memory, and which still need to be loaded. And if you place each object on each layout, and then delete all unnecessary ones, then a lot of dirt will appear in a large project, both in the code and on the layouts.

![XDjCSN8PW-c](https://user-images.githubusercontent.com/50992459/197526424-14eb5f62-7323-4f0d-abdd-bbbf13a20ad8.jpg)

**Comment by AshleyScirra** [comment]
I can't tell what's happening from that comment. The reason we have bug report guidelines is we need all that information in order to investigate. I would suspect it's still just dynamic texture loads. If you still think it's a problem, please file a new issue.
